src/domains/point_set.jl

```diff
@@ -25,34 +27,44 @@ julia> PointSet([(rand(),rand()) for i in 1:100])
 ```
 """
 struct PointSet{T,N} <: SpatialDomain{T,N}
-  coords::Matrix{T}
+  coords::Vector{SVector{N,T}} 
 
   function PointSet{T,N}(coords) where {N,T}
     @assert !isempty(coords) "coordinates must be non-empty"
     new(coords)
   end
 end
 
-PointSet(coords::AbstractMatrix{T}) where {T} =
-  PointSet{T,size(coords,1)}(coords)
+function PointSet(coords::AbstractMatrix{T}) where {T}
+  N = size(coords, 1)
+  points = Vector{SVector{N,T}}()
+  for row in eachcol(coords)
+    push!(points,SVector{N,T}(row))
+  end
+  PointSet{T,N}(points)
+end
 
-PointSet(coords::AbstractVector{NTuple{N,T}}) where {N,T} =
-  PointSet([c[i] for i in 1:N, c in coords])
+function PointSet(coords::AbstractVector{NTuple{N,T}}) where {N,T}
+  points = Vector{SVector{N,T}}()
+  for row in coords
+    push!(points,SVector{N,T}(row))
+  end
+  PointSet{T,N}(points)
+end
 
-nelms(ps::PointSet) = size(ps.coords, 2)
+nelms(ps::PointSet) = size(ps.coords,1)
 
 function coordinates!(buff::AbstractVector{T}, ps::PointSet{T,N},
                       location::Int) where {N,T}
```

Comment:
Could you also rename this variable `location` to `ind`?

Reply:
Why is that? It seems a change over the whole code I guess to avoid confusion between location (coordinate) and ind (index). Please confirm. I have seen using location in many places.

Reply:
It is not needed, but every occurrence of `location` is being replaced in the codebase by `ind` anyways. At some point they will go away. I thought it would be simple to do it here already.

Reply:
In the past we used `location` to mean the integer index, it was always equivalent to `ind`. Nowadays it is just clearer to use `ind` everywhere because sometimes this index is used in the table of values as opposed to in the spatial domain.